README.md

```diff
@@ -1,2 +1,226 @@
-# set-assignee-reviewer-when-ready
-A GitHub action to set assignee(s) or reviewer(s) when a PR has finished all checks.
+# Set Assignee Reviewer when Ready
+
+A [GitHub Action](https://help.github.com/en/actions) to set assignee(s) or reviewer(s) when a PR has finished all checks.
+
+> Want to get notified about a PR **before** the checks have completed?
+>
+> _**No?**_ I don't either.
+
+[![GitHub Marketplace](https://img.shields.io/github/v/release/ChrisCarini/set-assignee-reviewer-when-ready?label=Marketplace&logo=GitHub)](https://github.com/marketplace/actions/set-assignee-reviewer-when-ready)
+[![GitHub Marketplace](https://img.shields.io/github/contributors/ChrisCarini/set-assignee-reviewer-when-ready?label=Contributors&logo=GitHub)](https://github.com/ChrisCarini/set-assignee-reviewer-when-ready/graphs/contributors)
+[![GitHub Marketplace](https://img.shields.io/github/release-date/ChrisCarini/set-assignee-reviewer-when-ready?label=Last%20Release&logo=GitHub)](https://github.com/ChrisCarini/set-assignee-reviewer-when-ready/releases)
+
+## Why?
+
+This GitHub action was created because [Dependabot](https://docs.github.com/en/code-security/dependabot) only provides the ability to either:
+
+- Set [assignees](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#assignees) and/or [reviewers](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#reviewers) upon Dependabot PR creation.
+- Not set assignees or reviewers on Dependabot PRs.
+
+I have my repos configured to auto-merge Dependabot PRs once all required checks have passed. I really **only** want to get requested to review a PR (or, assigned a PR) if one of the required checks ends in an 'unacceptable' state (ie, it failed, or timed out). As such, this GitHub Action was created to enable me to be notified (via being requested for reviews) when a required check has failed. I implemented this GitHub Action with various configuration options to allow for flexibility in configuration; see the [options section](#-options) for details.
```

Comment:
It may be better to remove the first person prose from this in favor of just describing what this action is intended to do.

Reply:
sgtm!